src/server.rs

```diff
@@ -577,82 +577,143 @@ impl RollupBoostServer {
         payload_id: PayloadId,
         version: Version,
     ) -> RpcResult<OpExecutionPayloadEnvelope> {
+        // Start fetching L2 payload early (we'll always need it)
```

Comment:
Unless a tokio task is spawned under the hood this future will only begin resolving once it's `.await`ed on. Consider wrapping it in a `tokio::spawn` to start resolving immediately

Reply:
It's being joined below though, so that should be fine.

Reply:
true, but then there's no difference between this and spawning this future at the "join site". Plus the comment is misleading